src/classes/HH_OppContactRoles_TDTM.cls

```diff
@@ -283,31 +295,37 @@ public class HH_OppContactRoles_TDTM extends TDTM_Runnable {
     }
     
     /*******************************************************************************************************
-    * @description recreates Soft Credit OCR's for all Opportunities for the specified Household Accounts.
+    * @description Recreates Soft Credit OCR's for all Opportunities for the specified Household Accounts.
     * Used when Account Merge occurs.
-    * @param listHHId the list of Household Accounts 
-    * @return null
+    * @param hhIds The list of Household Account Ids
+    * @return void
     ********************************************************************************************************/
-    public static void recreateHouseholdSoftCredits(list<ID> listHHId) {
-    
-        // find all Opps for the Accounts
-        list<Opportunity> listOpp = [select Id, AccountId from Opportunity where AccountId in :listHHId];
+    public static void recreateHouseholdSoftCredits(List<Id> hhIds) {
+        List<Opportunity> opps = [
+            SELECT Id, AccountId, DisableContactRoleAutomation__c 
```

Comment:
I can't remember if we spoke about this one but it looks like we are querying for the field here but then not using it. Do we want to delete the existing ocrs when automation is disabled?

Reply:
To recap our discussion of our review of this code, it looks like this method is called only by the account merge and contact merge triggers. We have the option to:
1. Do not delete the existing ocrs if automation is disabled
2. Continue to delete the existing ocrs but ensure the household are regenerated in this scenario and create a unit test to verify
3. Intelligently check the ocrs for duplicate household roles and remove the duplicates only

Reply:
Even though the field is not used in the current method, Opportunities are passed to the household TDTM `run()` which expects `DisableContactRoleAutomation__c` field.
```
HH_OppContactRoles_TDTM hhocr = new HH_OppContactRoles_TDTM();
DmlWrapper dmlWrapper = hhocr.run(opps, null, TDTM_Runnable.Action.AfterInsert, Schema.SObjectType.Opportunity);      
```

Reply:
Thanks for raising question @jjbennett ! The expected behaviour has been confirmed to be the same regardless of the `Disable Contact Role Automation` since Contact/Account merging is different than the automation around creation.

Reply:
Thank you for confirming, no other questions. This looks good!